src/init/features/frameworks/index.ts

```diff
@@ -86,11 +80,73 @@ function toStack(
 ): Omit<Stack, StackOutputOnlyFields> {
   return {
     name: stackId,
-    codebase: { repository: cloudBuildConnRepo.name, rootDirectory: "/" },
     labels: {},
   };
 }
 
+/**
+ * Creates stack if it doesn't exist.
+ */
+export async function getOrCreateStack(projectId: string, setup: any): Promise<Stack | undefined> {
+  const location: string = setup.frameworks.region;
+  const deployMethod: string = setup.frameworks.deployMethod;
+  try {
+    return await getExistingStack(projectId, setup, location);
+  } catch (err: unknown) {
+    if ((err as FirebaseError).status === 404) {
+      logger.info("Create new stack");
+      if (deployMethod === "github") {
+        const cloudBuildConnRepo = await repo.linkGitHubRepository(
+          projectId,
+          location,
+          setup.frameworks.serviceName
+        );
+        const stackDetails = toStack(cloudBuildConnRepo, setup.frameworks.serviceName);
+        return await createStack(projectId, location, stackDetails);
+      }
+    } else {
+      throw new FirebaseError(
+        `Failed to get or create a stack using the given initialization details: ${err}`
+      );
+    }
+  }
+
+  return undefined;
+}
+
+async function getExistingStack(projectId: string, setup: any, location: string): Promise<Stack> {
+  let stack = await gcp.getStack(projectId, location, setup.frameworks.serviceName);
+  while (stack) {
+    await promptOnce(
+      {
+        name: "existingStack",
+        type: "input",
```

Comment:
Please use `type: confirm` instead to take care of paring 'yes' 'y', etc. See example uses in other parts of the codebase.